### PR TITLE
Disable stringWithFormat for nonstandard formats

### LINKED ISSUE #898: NSString.NSStringTests fails on OSX
I somehow missed this on the initial pass.

[ RUN      ] NSString.NSStringTests
NSStringTests.m:101: Failure
Value of: actualString
  Actual: ffffffffffffffff 2147483648
Expected: @"ffffffff -2147483648"
Which is: ffffffff -2147483648
[  FAILED  ] NSString.NSStringTests (1 ms)

The below lines are the culprit:

    actualString = [NSString stringWithFormat:@"%zx %zd", SIZE_MAX, INT_MIN];
    ASSERT_OBJCEQ(@"ffffffff -2147483648", actualString);

    actualString = [NSString stringWithFormat:@"%tx %td", PTRDIFF_MAX, PTRDIFF_MIN];
    ASSERT_OBJCEQ(@"7fffffff -2147483648", actualString);

For some reason, OSX prints a 64-bit value for zx even though the z modifier should restrict it to size_t... I'm also not clear on why INT_MIN is passed as zd, since its signed value would be invalid.

### PR DESCRIPTION
**Pull request by aballway**
While size_t and ptrdiff_t are 32 bit on windows, they are 64 on the reference platform, causing the test to fail.

Fixes #898

### CODE REVIEW

File: tests/unittests/Foundation/NSStringTests.m [hunk 1]
```diff
@@ -78,51 +78,54 @@ void testUrlCharacterSetEncoding(NSString* decodedString, NSString* encodedStrin
...
+// size_t and ptrdiff_t are not a set size causing this to fail on the reference platform
+#ifdef WINOBJC
     actualString = [NSString stringWithFormat:@"%zx %zd", SIZE_MAX, INT_MIN];
-    ASSERT_OBJCEQ(@"ffffffff -2147483648", actualString);
+    EXPECT_OBJCEQ(@"ffffffff -2147483648", actualString);
 
     actualString = [NSString stringWithFormat:@"%tx %td", PTRDIFF_MAX, PTRDIFF_MIN];
-    ASSERT_OBJCEQ(@"7fffffff -2147483648", actualString);
+    EXPECT_OBJCEQ(@"7fffffff -2147483648", actualString);
+#endif
```
Comment (ms-jihua) [inline tests/unittests/Foundation/NSStringTests.m]: hmm, is it worth adding an else here just so we can document how %z_ and %t_ behave on the reference platform?

**Comment by ms-jihua** [comment]
Thanks for taking care of this for me!

**Comment by ms-jihua** [comment]
:shipit:

File: tests/unittests/Foundation/NSStringTests.m (hunk 1 above)
Comment (aballway) [inline tests/unittests/Foundation/NSStringTests.m]: I don't think so, feels a bit silly to document what it isn't

**Comment by ms-jihua** [comment]
Oh, actually, since this is the last failing test (!), can you update the doc?

---
In reply to: [250304213](https://github.com/Microsoft/WinObjC/pull/1044#issuecomment-250304213) [](ancestors = 250304213)

File: docs/Testing/OSXWinObjCTests.md [hunk 2]
```diff
@@ -50,136 +50,4 @@ While it's possible to checkout a separate git enlistment on an OSX machine and
...
-    * [  FAILED  ] NSHTTPURLResponse.SuggestedFilename_3
-    * [  FAILED  ] NSHTTPURLResponse.SuggestedFilename_4
-    * [  FAILED  ] NSHTTPURLResponse.canBeArchived
-    * [  FAILED  ] NSString/StringsFormatPropertyList.CanDeserialize/0, where GetParam() = L"\xFEFFkey1=value1;\n\"key2\"=\"value2\";"
-    * [  FAILED  ] NSString/StringsFormatPropertyList.CanDeserialize/1, where GetParam() = L"key1=value1;\n\"key2\"=\"value2\";"
-    * [  FAILED  ] NSString/ComparisonTests.PrefixSuffix/8, where GetParam() = (, , '\x1' (1), '\0', '\x1' (1), '\x1' (1), '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0')
-
-    * 126 FAILED TESTS
-    *   YOU HAVE 9 DISABLED TESTS
+There are no known failures in Foundation.UnitTests
```
Comment (rajsesh-msft) [inline docs/Testing/OSXWinObjCTests.md]: >There [](start = 0, length = 5)

Woot!

**Comment by rajsesh-msft** [comment]
:shipit:

**Pull request merged**